### PR TITLE
Support apocryphal books

### LINKED ISSUE #612: Support deuterocanonical / apocryphal books

### PR DESCRIPTION
**Pull request by tobias-klein**
This PR fixes #612.

To test this, install one of the following modules:

* KJVA (English)
* engRV1895eb (English)
* engasvbt2021eb (English)
* Vulgate (Latin)

Note that this PR adds a database migration which will modify your existing database. If you do not want your production database to be modified during testing - work with a test user.

Behavior:
* If a module does not have any apocryphal books, the complete section will be hidden.
* If specific apocryphal books are not present, the links are not appearing (for regular books the links are gray if the book is not present).

image

### CODE REVIEW

**Review by zhuiks — CHANGES_REQUESTED** [review]
Nice!

I notice a few things on the UI side:

- Somehow I can switch from translation with apocryphal books to canonical only translation:
Screenshot from 2022-02-09 19-44-55

- Seems like headers are not always displayed correctly:
Screenshot from 2022-02-09 19-58-00

Let me know if I need to provide more concrete steps to reproduce.

File: app/frontend/helpers/sword_module_helper.js [hunk 1]
```diff
@@ -176,6 +176,21 @@ module.exports.moduleHasHeaders = async function(moduleId) {
   }
 };
 
+module.exports.moduleHasApocryphalBooks = async function(moduleId) {
+  const books = await ipcNsi.getBookList(moduleId);
+
+  for (let i = 0; i < books.length; i++) {
```
[the remark this answers is not in the document]
Comment (tobias-klein) [reply app/frontend/helpers/sword_module_helper.js:182]: Good point! This is more significant on Android! I will look into it.

**Comment by tobias-klein** [comment]
> * Seems like headers are not always displayed correctly:

This was a bug in node-sword-interface, but I had fixed it. Did you run `npm install` after switching to the branch?

**Comment by tobias-klein** [comment]
> * Somehow I can switch from translation with apocryphal books to canonical only translation:

This bug was already existing before. When you have opened a book in the OT for example, it is still possible to switch to a translation that does not contain OT books. I have now opened issue #616 for it.

File: app/frontend/helpers/sword_module_helper.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (tobias-klein) [reply app/frontend/helpers/sword_module_helper.js]: This is fixed now.

**Comment by tobias-klein** [comment]
@zhuiks Can you test once more with the latest changes (ensure to run `npm install` as well) and then approve if everything is fine?

**Review by zhuiks — APPROVED** [review]
Thank you! Works very nice

**Pull request merged**